.github/workflows/tests.yml

```diff
@@ -35,6 +35,15 @@ jobs:
           python-version: 3.9
       - name: Test Setup
         run: python -m pip install tox
+      - name: Free up disk space on Linux
+        if: runner.os == 'Linux'
+        run: |
+          df -h
+          sudo swapoff -a
+          sudo rm -f /swapfile
+          sudo apt clean
+          docker rmi $(docker image ls -aq)
```

Comment:
Will that not slow down the next runs as images need to be re-pulled?

Reply:
yep I think so, but I'm not sure how these runners get reused, though it seems they do in some fashion, since they were running out of disk

Reply:
The images it seemed to delete were not ours
```
Untagged: buildpack-deps:buster
Untagged: buildpack-deps@sha256:b2c2267e1944a42dcdd448b66e8c8c889a3f5d1ab9705421326077a40a8aa0bc
Deleted: sha256:7ceeb7ee443c80095761b84c6e6d14fe95695139a577f5896d82df863c05d739
Deleted: sha256:fc669e37da096189fcdf5c8fb7547a6b78f000909e5be0b0bb56c11664fa03b8
Deleted: sha256:bce6a22b04d542388c8a67[29](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:30)4a7fb7a167bfc8eaa580c2215c15d61eeb45198d
Deleted: sha256:7e85d254ef9592810a3e3e4bf7b064a4f97907a139c2f6c7016667d0c13e936f
Untagged: buildpack-deps:bullseye
Untagged: buildpack-deps@sha256:593db75ce93184b16d45021566ed2d77f5009a8ce88da8254919a2e9e6ec14bf
Deleted: sha256:af4a6a6182dd5c7c765fa6e5894539d38a174d2c864885c42e3b8d404b289cbe
Deleted: sha256:2298b1e7fdf1b8fe141ee4d7cd39d58a45cdcd984890627a919cd555bded1d17
Deleted: sha256:4c8b85c423d73278fc5ed6ac818f92b0e791cb0c5b7b3c32a946a6112df4b4bf
Deleted: sha256:c878afa097c72062fb9ae3872d392e1826a66d5c1f500172ff09fa8a883117ab
Untagged: debian:10
Untagged: debian@sha256:7cd85d3d51a435062010581f14c5e3f9428388ac7234cc9a1e23dd14d7e4e914
Deleted: sha256:1df4474660a180f2a805d8477b78f9774d[30](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:31)2afb2d[31](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:32)118106bffc0cb144e910
Deleted: sha256:15f0e9f7640e22f69970115f1c090c64fa363a3f3a688c7a04c1a4eefd0ce48f
Untagged: debian:11
Untagged: debian@sha256:630454da4c59041a2bca987a0d54c68962f1d6ea37a3641bd61db42b75[32](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:33)34f2
Deleted: sha256:35073ea3b0b7ab55ea55e1a111a8ded11a58ef2b6b3ae0e706160520e612df6f
Deleted: sha256:f370a05d981b3c9d2cbc1253d381ad6a5b0[33](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:34)8e2a4bc3f29e8bc888929cedca4
Untagged: node:18
Untagged: node@sha256:5b0596963c8653bac197457d0da388ac603658011f13cb15d303a54bb83679fa
Deleted: sha256:0f9df951673ddc5c6a4b0b25e8d58248c93a6e1bcbf8207f2650b97492c08201
Deleted: sha256:7c464dc3[34](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:35)40b452005[35](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:36)8a9073c24330d5af53072181f471576d5d9240ba32e
Deleted: sha256:2c7a68520c506c76c051177e8cc496abe5a1799f[36](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:37)357eb56401345b885914ed
Deleted: sha256:64595435a9b864b443bd298fe7463f9f0f90e9b80992f7881dd6212b12c290a1
Deleted: sha256:c[37](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:38)67412dc7a4b61d34c21949c7b277c007d7ee979170ae088b30604b394916d
Deleted: sha256:47be7118de812a5e4dd0c807[38](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:39)586f76c4171587359b70e862b1760e93d0b18b
Deleted: sha256:b80181f01d740ba89ac4b6ce18ec3ef3f374f14d972c2f874813c3[39](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:40)90687e22
Deleted: sha256:8faba6[40](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:41)c5ba02612700e22f88797a571828443075845780f431d956aa72c456
Deleted: sha256:61581d479298c795fa3cfe95[41](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:42)9a5cec510085ec0d040306f69e491a598e7707
Untagged: node:18-alpine
Untagged: node@sha256:93d91deea65c9a0475507e8bc8b1917d6278522322f00c00b3ab09cab6830060
Deleted: sha256:9112cb38e57e8bfbc18330dfe346483e20db1fc77c84e[42](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:43)38f63dd6fef798f66
Deleted: sha256:80e922485b4c3ddeea34eaa0574c4de81191e80105194a6e3e0487b3c7b24ec7
Deleted: sha256:4ef1d1625f6b8953f1cd7611bb074237f51cd62cac[43](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:44)ec09ce4ed8d06e019b76
Deleted: sha256:a6d6d085ae1a11d3e1822b7335ac5c9139cd9a4f3808f266eae63e4c7f942147
Untagged: moby/buildkit:latest
Untagged: moby/buildkit@sha256:e70e7a9bfaeafd2cb101b050ca78e1f73032473fc0aac5b10f35c5d26df1e2c6
Deleted: sha256:869c7038a926857232869ac4637d79d7a0527823340c5431caeeec66a3ce7ec6
Deleted: sha256:ec5171853e0b9294fcf1[44](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:45)3ac18bf0a60652ebef0bf17b9889823121e181804d
Deleted: sha256:a659db21cd3eb912b7cfb506[45](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:46)d8a2faedfe3701b5ea604a0927e0ea569c96c3
Deleted: sha256:24d34f03896e9277fc99500af9ab2b07f06c22e96620417b74e265630ce6db63
Untagged: node:16
Untagged: node@sha256:c83c7fa87b7449675b422877f547233f3d05ac75a4363ed030fc3635dd9cc395
Deleted: sha256:81c6e5ef93fcc0f603ac62cb5b8fbda4575af20f5003d5da1d229b54d02d97c4
Deleted: sha256:6a4ba61688a53c2bb86d9cc1359f3c9f6ee0e4b550358d4dc52e6ba096f6c3b9
Deleted: sha256:8fdda38f2ccae25e38de6ccac9cc91de2ee9990275d0c13689ba09599a15703f
Deleted: sha256:798635aaff594d4dca26a7d606ab7854ad99c8e6cd0b35a98aedbccce0d5849e
Deleted: sha256:83c9e27bb29e7a06652c997147dd287ea2cc7e59bd1fd2af7a77b376b74b4e7a
Deleted: sha256:7cd4d1040ad090090eeddbb89a38703a81b9455042bf016619821637c300d1a8
Deleted: sha256:819cf82e721ee294390e9fb606c092dbc4e22df9a217dbac194b4aa459bd91c0
Deleted: sha256:cf90d3440eb8324b02cabe0ccc0d51b67c5bc94474a26e64353ec4260837da00
Deleted: sha256:0c23ed2428e1ef41fdb571789f6968dd47c95135311b1c8bf56[46](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:47)2c98581694b
Untagged: ubuntu:20.04
Untagged: ubuntu@sha256:c9820a44b950956a790c35[47](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:48)00c1166a7ec6[48](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:49)bc0d215fa438d3a339812f1d01
Deleted: sha256:14be0685b7682b182af5b862c9638cb1cb4ca1a70bd5aa90deed96e9cca881e7
Deleted: sha256:f5bb4f853c84f6579d86f72de5f1d43fc8866443d9a027e2106e3755acefd61c
Untagged: node:16-alpine
Untagged: node@sha256:6c381d5dc2a11dcdb693f0301e8587e43f440c90cdb8933eaaaabb905d44cdb9
Deleted: sha256:d22e42cbfdbbd3264cc7dc26ad90192ed40027ba7f93cdba22b87[49](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:50)61b8e27a3
Deleted: sha256:8db90ef907d9ebdd9b332cc5e01d71bb1267da20267eaf537d19c24784fd1615
Deleted: sha256:9d8557d8b2655cc265943db51f176a6f2da4cb21328f78b47655c31ccd7c028e
Deleted: sha256:3a71837eb0412ca7a0e3cff75fe53609c924794249f89ac9a1347ed801601535
Untagged: alpine:3.16
Untagged: alpine@sha256:cbe5d5973103a2d03408d1689a6efde4ea4920bde9f4b51fe7872e60ce2d8e56
Deleted: sha256:5cb2da5c2391f597316eafada2ec327caf3f0ce8df1022f39273d2e73d7d002c
Deleted: sha256:18eb8b5891f2056b0a6c9978359916a519e8fdeec08c13c6383b922cd15fcfb2
Untagged: alpine:3.17
Untagged: alpine@sha256:e95676db9e4a4f16f6cc01a8915368f82b018cc07aba951c1bd1db586c081388
Deleted: sha256:1f73a9d63274924f077b22c7e5a894667cb6a7073638e7d5df3b43be14d1c9ab
Deleted: sha256:3dab9f8bf2d28c8bd1047f3ac2d0c72f3570562f491e67ef8179dfdcc68bccff
Untagged: alpine:3.18
Untagged: alpine@sha256:82d1e9d7ed48a7523bdebc18cf6290bdb97b82302a8a9c27d4fe885949ea94d1
Deleted: sha256:c1aabb73d2339c5ebaa3681de2e9d9c18d5748[50](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:51)45a4e311d9f8004bec208d67
Deleted: sha256:78a822fe2a2d2c84f3de4a403188c45f623017d6a4521d23047c9fbb0801794c
Untagged: ubuntu:18.04
Untagged: ubuntu@sha256:152dc042452c496007f07ca9127571cb9c29697f42acbfad72324b2bb2e43c98
Deleted: sha256:f9a80a55f492e823bf5d[51](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:52)f1bd5f87ea3eed1cb31788686aa99a2fb61a27af6a
Deleted: sha256:548a79621a426b4eb077c926eabac5a8620c454fb230640253e1b44dc7dd7562
Untagged: node:14
Untagged: node@sha256:a158d3b9b4e3fa813fa6c8c590b8f0a860e015ad4e59bbce5744d2f6fd8461aa
Deleted: sha256:1d12470fa662a2a5cb50378dcdc8ea228c1735747db410bbefb8e2d9144b54[52](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:53)
Deleted: sha256:d7f4fb376c48badb9879c97df87580480be3f4dcad2e2f33425792d164933fa3
Deleted: sha256:71fe964a0bdce3b1795d90eb1f8a501dd719e8612faeb40783706c962339c75c
Deleted: sha256:559303f34408584d79e77eab1ec15eb80d04d7[53](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:54)a87a38c8da31[54](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:55)2440c0a64a
Deleted: sha256:483ca60b49ed280253[55](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:56)f381d9e2b759e67e66b9dfb266c66f57acf0cb47033e
Deleted: sha2[56](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:57):f1edb12649962bbc5048b132c2201e98d474b9f99e6936d41633b43d6e7bd91d
Deleted: sha256:9f9d995a1674f94b553c90e4eb3502db04a8220403f3add46486659752a3a108
Deleted: sha256:a61e818f0b1f7522e3a27211026f564f1fd98ae2034229d1233623e92a6a7961
Deleted: sha256:16c88c4e67dd8ed4c67c0e9c1dae7df697f4607eab4c3d2f12b1b18206a6a6e4
Deleted: sha256:b2dba74777543b60e1a5be6da44e67659f51b8df1e96922205a5dde6b92dda3c
Untagged: node:14-alpine
Untagged: node@sha256:434215b487a329c9e867202ff89e704d3a75e554822e07f3e0c0f9e606121b33
Deleted: sha256:0dac3dc27b1ad[57](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:58)0e6c3a7f7cd29e88e7130ff0cad31b2ec5a0f222fbe971bdb
Deleted: sha256:3f3a6e7f9c9d353391c7[58](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:59)99b353f0f14c203e98fdbd0fd11bf6b84a74dab522
Deleted: sha256:dee85e6b7f481bd5bbdff9fdfd8568ee70adb0d277f352348ef099f52afb46df
Deleted: sha256:d6c83f14fe35dff122c995ada9da84a33ad2b7bf357133c3fa4c2baea3d3755b
Deleted: sha256:f1417ff83b319fbdae6dd9cd6d8c9c88002dcd75ecf6ec201c8c6894681cf2b5
Untagged: buildpack-deps:stretch
Untagged: buildpack-deps@sha256:78e995165a5788c2f55aed6e548d8f6c1534830d4310c870408fccb2da8c5b2e
Deleted: sha256:9d4f40bcbdd3f7d03a965637bd7751168ba03fbd5758d48688a2cb91884cb6f5
Deleted: sha256:ee88b03cb842e6338e417fa5f15cc04354f[59](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:60)42c84966a36a13525cef96ff6ef
Deleted: sha256:7c2fdbf5468f262579288[60](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:61)d6ef6faf0850d6fd5551b[61](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:62)ee0b9f4a43c05f46df
Deleted: sha256:941361304949d91ebee5a0663c09f[62](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:63)c240d78061ffca87a6ebd1c72fc96406b
Deleted: sha256:8282b75f66b9ebefafdb17440bfe8bd65fbd619b229c4afcd2ec8a354dcce377
Untagged: debian:9
Untagged: debian@sha256:c5c5200ff1e9c73ffbf188b4a67eb1c91531b644856b4aefe86a58d2f0cb05be
Deleted: sha256:662c05203bab4c59568d24689fa5c3955439360a35c483178598d226b9a5ad10
Deleted: sha256:47dde6f41d1fcca77df950cac35c7103996a9333244a7406248c9b6898215469
Untagged: ubuntu:16.04
Untagged: ubuntu@sha256:1f1a2d56de1d604801a9671f301190704c25d604a416f59e03c04f5c6ffee0d6
Deleted: sha256:b6f50765242581c887ff1acc2511fa2d885c52d8fb3ac8c4bba131fd86567f2e
Deleted: sha256:0214f4b057d78b44fd12702828152f67c0ce115f9346acc[63](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:64)acdf997cab7e7c8
Deleted: sha256:1b9d0485372c5562fa614d5b357[66](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:67)f6c442539bcee9825a6e90d1158c3299a61
Deleted: sha256:3c0f34be6eb98057c607b9080237cce0be0b86f52d51ba620dc018a3d421baea
Deleted: sha256:be96a3f634de79f523f07c7e4e0216c28af45eb5776e7a6238a2392f71e010[69](https://github.com/gretelai/gretel-python-client/actions/runs/5802379649/job/15728658694#step:5:70)
```
and these don't use the on prem runners